code/src/components/Item.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import styled from 'styled-components/macro';
+import { useDispatch } from 'react-redux';
+import moment from 'moment';
+
+import { todos } from '../reducers/todos';
+
+export const Item = ({ id, text, isCompleted, timestamp }) => {
+	const dispatch = useDispatch();
+	const [completed, setCompleted] = useState(false);
+	const handleCompleted = (id) => {
+		dispatch(todos.actions.toggleCompleted(id));
+	};
+	const handleDeleteItem = (id) => {
+		dispatch(todos.actions.removeItem(id));
+	};
+	return (
+		<ItemContainer style={{ opacity: `${isCompleted ? '0.3' : '1'} ` }}>
+			<ItemText>{text}</ItemText>
+			<p>Added: {moment(timestamp).format('MMM Do YY')}</p>
+			<CompleteDeleteContainer>
+				<Button
+					onClick={(e) => {
+						setCompleted(id);
+						handleCompleted(id);
+					}}>
+					Click to complete
+				</Button>
+				<Button onClick={() => handleDeleteItem(id)}>Delete</Button>
+			</CompleteDeleteContainer>
+		</ItemContainer>
+	);
+};
+
+const ItemContainer = styled.div`
```

Comment:
Set a width on this so that it's not dynamic depending on the length of the todo